Enable PCIe with ComboPHY on Radxa Cubie A5E

Based on Armbian patch by Marvin Wewer, adapted for U-Boot v2026.04.

--- a/dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts
+++ b/dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts
@@ -63,6 +63,30 @@
 		gpio = <&r_pio 0 8 GPIO_ACTIVE_HIGH>;	/* PL8 */
 		enable-active-high;
 	};
+
+	reg_pcie_vcc3v3: regulator-pcie-vcc3v3 {
+		compatible = "regulator-fixed";
+		regulator-name = "pcie-pwren";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&r_pio 0 11 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		regulator-always-on;
+		regulator-boot-on;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pcie_pwren_pins>;
+	};
+
+	reg_vcc_3v3: vcc-3v3 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc-3v3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&r_pio 0 7 GPIO_ACTIVE_HIGH>;
+		regulator-always-on;
+		regulator-boot-on;
+		enable-active-high;
+	};
 };
 
 &ehci0 {
@@ -125,6 +149,15 @@
 	vcc-pi-supply = <&reg_cldo3>;
 	vcc-pj-supply = <&reg_cldo4>;
 	vcc-pk-supply = <&reg_cldo1>;
+
+	pcie0_pins_ph: pcie0-ph {
+		pins = "PH19";
+		function = "pcie0";
+		allwinner,pinmux = <6>;
+		drive-strength = <20>;
+		bias-pull-up;
+		power-source = <3300>;
+	};
 };
 
 &r_i2c0 {
@@ -318,6 +351,14 @@
  *	vcc-pl-supply = <&reg_aldo3>;
  */
 	vcc-pm-supply = <&reg_aldo3>;
+
+	pcie_pwren_pins: pcie-pwren-pins {
+		allwinner,pins = "PL11";
+		function = "gpio_out";
+		allwinner,pinmux = <1>;
+		drive-strength = <10>;
+		bias-disable;
+	};
 };
 
 &uart0 {
@@ -326,6 +367,21 @@
 	status = "okay";
 };
 
+
+&combophy {
+	select3v3-supply = <&reg_vcc_3v3>;
+	status = "okay";
+};
+
+&pcie {
+	reset-gpios = <&pio 7 11 GPIO_ACTIVE_HIGH>;
+	wake-gpios = <&pio 7 12 GPIO_ACTIVE_HIGH>;
+	num-lanes = <1>;
+	slot-3v3-supply = <&reg_pcie_vcc3v3>;
+	switch-sel-gpios = <&pio 1 6 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
 &usb_otg {
 	/*
 	 * The USB-C port is the primary power supply, so in this configuration
